sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
It's the result of Window test query. For the Windows query, there were nested Projects.
```
test("broadcast hint with window") {
    checkSQL(
      """
        |SELECT /*+ MAPJOIN(parquet_t1) */
        |       x.key, MAX(y.key) OVER (PARTITION BY x.key % 5 ORDER BY x.key)
        |FROM parquet_t1 x JOIN parquet_t1 y ON x.key = y.key
      """.stripMargin,
      "broadcast_hint_window")
  }
```

I had the same feeling why some "SELECT" doesn't happen. 

After https://issues.apache.org/jira/browse/SPARK-16576 , I think this kind of weirdness will be reduced.